core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -397,9 +398,35 @@ abstract class AppBase(
   }
 
   protected def postCompletion(): Unit = {
+    updateAttemptIdTracker()
     calculateAppDuration()
   }
 
+  /**
+   * Updates the appAttemptIdTracker with the attemptId of the app, if
+   * the attemptId is greater than the current attemptId.
+   * This is used to determine the newest attemptId for an app.
+   */
+  private def updateAttemptIdTracker(): Unit = {
+    // If appMetaData (hence appId) is defined
+    if (appMetaData.isDefined) {
+      // Note: `compute` is atomic and thread-safe.
+      AppBase.appAttemptIdTracker.compute(appId, (_, currentAttemptId) => {
+        Option(currentAttemptId) match {
+          case Some(id) if this.attemptId > id => this.attemptId
+          case None => this.attemptId
+          case _ => currentAttemptId
+        }
+      })
+    }
+  }
+
+  lazy val attemptId: Int = sparkProperties.getOrElse("spark.app.attempt.id", "1").toInt
```

Comment:
After confirming that the index is not 0-based, please add a comment so that anyone reading the code won't be confused if it is set based on technical reason in tools code, or because Spark first attempt is 1.